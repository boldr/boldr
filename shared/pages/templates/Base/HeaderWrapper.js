/* @flow */
import React, { Component } from 'react';
import { push } from 'react-router-redux';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import type { ReactElement, ReactChildren } from '../../../types/react';
import PrimaryHeader from '../../../components/PrimaryHeader';
import { logout } from '../../../state/modules/account/actions';
import { selectSetting, getSettings } from '../../../state/modules/boldr/settings';

type Props = {
  header: ReactElement,
  helmetMeta?: ReactElement,
  hero?: ReactElement,
  children: ReactChildren,
  footer?: ReactElement,
  navigate: Function,
  actions: Object,
  ui: Object,
  menu: Object,
  settings: Object,
  logo: Object,
  auth: Object,
};

const mapStateToProps = (state: Object) => {
  return {
    boldr: state.boldr,
    settings: state.boldr.settings,
    logo: selectSetting(state, 'site_logo'),
    auth: state.account.auth,
    menu: state.boldr.menu.main,
    ui: state.boldr.ui,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    actions: bindActionCreators({ logout, pushState: push }, dispatch),
    navigate: (url) => dispatch(push(url)),
    dispatch,
  };
};

@connect(mapStateToProps, mapDispatchToProps)
class HeaderWrapper extends Component {
  constructor() {
    super();
    (this: any).handleLogoClick = this.handleLogoClick.bind(this);
    (this: any).handleLogoutClick = this.handleLogoutClick.bind(this);
    (this: any).handleDashClick = this.handleDashClick.bind(this);
    (this: any).handleProfileClick = this.handleProfileClick.bind(this);
    (this: any).handlePrefClick = this.handlePrefClick.bind(this);
  }

  handleLogoClick = (e) => {
    this.props.navigate('/');
  }
  handleDashClick = (e) => {
    this.props.navigate('/admin');
  }
  handleProfileClick = (e) => {
    this.props.navigate('/profile');
  }
  handlePrefClick = (e) => {
    this.props.navigate('/account/preferences');
  }
  handleLogoutClick = (e) => {
    this.props.actions.logout();
  }

  props: Props;

  render() {
    return (
      <PrimaryHeader
        auth={ this.props.auth }
        logo={ this.props.logo }
        settings={ this.props.settings }
        menu={ this.props.menu }
        handleLogoClick= { this.handleLogoClick }
        handleLogoutClick={ this.handleLogoutClick }
        handlePrefClick={ this.handlePrefClick }
        handleProfileClick={ this.handleProfileClick }
        handleDashClick={ this.handleDashClick }
      />
    );
  }
}
export default HeaderWrapper;
