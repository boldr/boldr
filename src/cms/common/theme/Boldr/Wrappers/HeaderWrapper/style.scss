.boldr__theme-header {
  margin-bottom: 10px;
}
