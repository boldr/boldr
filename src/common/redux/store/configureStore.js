import { createStore, applyMiddleware, compose } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { routerMiddleware } from 'react-router-redux';
import createLogger from 'redux-logger';
import promiseMiddleware from 'redux-promise';
import logger from '../middleware/logger';
import rootReducer from '../modules';

/*
 * @param {Object} initial state to bootstrap our stores with for server-side rendering
 * @param {History Object} a history object. We use `createMemoryHistory` for server-side rendering,
 *                          while using browserHistory for client-side
 *                          rendering.
 */
export default function configureStore(initialState, history) {
  const middleware = [thunkMiddleware, promiseMiddleware, routerMiddleware(history)];
  if (__DEV__) {
    middleware.push(createLogger());
  }

  const store = createStore(rootReducer, initialState, compose(
    applyMiddleware(...middleware),
    typeof window === 'object' && typeof window.devToolsExtension !== 'undefined' ? window.devToolsExtension() : f => f
  ));

  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('../modules/index', () => {
      const nextReducer = require('../modules/index').default;
      store.replaceReducer(nextReducer);
    });
  }

  return store;
}
