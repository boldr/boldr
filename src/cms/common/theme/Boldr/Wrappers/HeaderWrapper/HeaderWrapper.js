import React from 'react';
import './style.scss';

const HeaderWrapper = (props) => {
  return (
    <header className="boldr__theme-header">
      { props.children }
    </header>
  );
};

export default HeaderWrapper;
