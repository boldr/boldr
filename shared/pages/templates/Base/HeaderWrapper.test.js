import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import HeaderWrapper from './HeaderWrapper';

const middlewares = [];
const mockStore = configureStore(middlewares);

describe('<BaseTemplate />', () => {
  const initialState = {};
  const store = mockStore(initialState);
  const wrapper = shallow(<Provider store={ store }><HeaderWrapper /></Provider>);
  test('renders <HeaderWrapper /> without breaking', () => {
    expect(wrapper.find('div').length).toBe(0);
  });
  test('renders snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
