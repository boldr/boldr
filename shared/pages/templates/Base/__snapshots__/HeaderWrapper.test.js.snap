exports[`<BaseTemplate /> renders snapshot 1`] = `<Connect(HeaderWrapper) />`;
